src/sp/protocol/reqrep0/nano_tcp.c

```diff
@@ -289,6 +290,12 @@ nano_ctx_send(void *arg, nni_aio *aio)
 	nni_mtx_lock(&p->lk);
 	if (!p->busy) {
 		p->busy = true;
+		pub_extra *pub_extra_info =
+		    (pub_extra *) nni_aio_get_prov_extra(aio, 0);
+		if (pub_extra_info) {
+			nni_aio_set_prov_extra(
+			    &p->aio_send, 0, pub_extra_info);
+		}
```

Comment:
we should also consider the case when pipe is busy. otherwise memleak would happen

Reply:
Fixed, please review the last change;